src/pages/dagens-transportmiddel/DagensTransportmiddel.tsx

```diff
@@ -30,65 +39,133 @@ const DagensTransportmiddel = (): ReactElement => {
     dagensTransportmiddelValidert, settDagensTransportmiddelValidert,
   } = useAppStore();
 
-  const handleKilometerChange = (nyInput: string) => {
-    settAntallKilometerState(nyInput);
-    settDagensTransportmiddelValidert(undefined);
+  const validerAntallKilometerInput = (): FeiloppsummeringFeil[] => {
+    if (dagensTransportMiddelEgenBilChecked) {
+      if (!validerNumerisk(antallKilometerState)) {
+        return [
+          {
+            skjemaelementId: antallKilometerSpørsmål.id,
+            feilmelding: 'Du må oppgi gyldig verdi for kilometer',
+          },
+        ];
+      }
+    }
+    /* Gyldig verdi skrevet inn,
+     * skal ikke validere dette feltet igjen før brukeren trykker for å gå videre:
+     */
+    settSkalViseKilometerFeil(false);
```

Comment:
Jeg har visst misforstått designsystemets guidelines, så denne linja her (og tilsvarende for månedlige utgifter) trengs ikke